pkg/catalogv2/system/system.go

```diff
@@ -238,36 +257,56 @@ func (m *Manager) Remove(namespace, name string) {
 	}
 }
 
+// install tries to install a new version of a chart.
+// If the exact version is provided, it will try to install it regardless of whether minVersion is provided.
+// If minVersion is provided on its own, it will try to install it only if the current version is earlier than
+// minVersion.
+// A failure to find a chart for a provided version leads to an error being thrown, without any change in state.
+// If no version is provided, it will try to install the latest version available.
+// If a release with the version to be installed is already installed, or is pending install, upgrade or rollback, this
+// does nothing.
 func (m *Manager) install(namespace, name, minVersion, exactVersion string, values map[string]interface{}, forceAdopt bool, installImageOverride string) error {
 	index, err := m.content.Index("", "rancher-charts", true)
 	if err != nil {
 		return err
 	}
 
-	v := ">=0-a" // latest - this is special syntax to match everything including pre-releases build
+	const latestVersionMatcher = ">=0-a" // latest - special syntax to match everything including pre-release builds
+
+	v := latestVersionMatcher
 	var isExact bool
+
 	if exactVersion != "" {
 		v = exactVersion
 		isExact = true
+	} else if minVersion != "" {
```

Comment:
We'd need to make sure this doesn't change the webhook's version pinning. As with fleet version, If both min and exact are provided, then min takes precedence. Hopefully, this doesn't change this and the periodic check for webhook version...

Reply:
Good point. My understanding is that, given how the webhook [is installed](https://github.com/rancher/rancher/blob/release/v2.7.11/pkg/controllers/dashboard/systemcharts/controller.go#L129), if both min and exact version are specified, installation will behave just as it does with Fleet. The same behaviour is implemented differently:
* Fleet: set exact version to min version, call `manager.Ensure` with `minVersion` set as that version, and empty exact version. For Fleet, we always call `Ensure` with an empty `exactVersion` parameter because we don't want to force downgrades if a more recent version is installed.
* Webhook: set exact version to an empty string, then call `manager.Ensure` with the min version and that exact version

The difference seems to be that if only an exact version is provided, Rancher will force a webhook downgrade if a more recent version is installed, whereas that will not happen for Fleet.